drivers/include/at24cxxx.h

```diff
@@ -0,0 +1,211 @@
+ */
+enum {
+    AT24CXXX_DEV_ADDR_00    = (0x50),   //A1 = 0, A0 = 0
+    AT24CXXX_DEV_ADDR_01    = (0x51),   //A1 = 0, A0 = 1
+    AT24CXXX_DEV_ADDR_10    = (0x52),   //A1 = 1, A0 = 0
+    AT24CXXX_DEV_ADDR_11    = (0x53)    //A1 = 1, A0 = 1
+};
+
+/**
+ * @brief struct that holds initialization parameters
+ */
+typedef struct at24cxxx_params {
+    i2c_t i2c;                      /**< I2C bus number */
+    uint8_t dev_addr;               /**< I2C device address */
+    gpio_t pin_wp;                  /**< write protect pin */
+    size_t eeprom_size;             /**< memory capacity */
+} at24cxxx_params_t;
+
+/**
+ * @brief struct that represents an AT24CXXX device
+ */
+typedef struct {
+    at24cxxx_params_t params;     /**< parameters */
+} at24cxxx_t;
+
+/**
+ * @brief Get address of default device parameters
+ *
+ * @return    default device parameters
+ */
+const at24cxxx_params_t *at24cxxx_get_default_dev_params(void);
+
+/**
+ * @brief   Initialize an AT24CXXX device handle with AT24CXXX parameters
+ *
+ * @param[in, out] dev    AT24CXXX device handle
+ * @param[in] params      AT24CXXX parameters
+ *
+ * @return    AT24CXXX_OK on success
+ * @return    -EINVAL if input paramters are NULL
+ */
+int at24cxxx_init(at24cxxx_t *dev, at24cxxx_params_t *params);
+
+/**
+ * @brief   Read a byte at a given position \p pos
+ *
+ * @param[in] dev      AT24CXXX device handle
+ * @param[in] pos      position in EEPROM memory
+ *
+ * @return    Read byte
+ */
+uint8_t at24cxxx_read_byte(const at24cxxx_t *dev, uint32_t pos);
+
+/**
+ * @brief Sequentially read \p len bytes from a given position \p pos
+ *
+ * errno is set to ERANGE if EEPROM bounds would be exceeded.
+ *
+ * @param[in] dev       AT24CXXX device handle
+ * @param[in] pos       position in EEPROM memory
+ * @param[out] data     read buffer
+ * @param[in] len       requested length to be read
+ *
+ * @return    \p len on success
+ * @return    0 on failure
+ */
+size_t at24cxxx_read(const at24cxxx_t *dev, uint32_t pos, uint8_t *data,
+                     size_t len);
+
+/**
+ * @brief   Write a byte at a given position \p pos
+ *
+ * @param[in] dev      AT24CXXX device handle
+ * @param[in] pos      position in EEPROM memory
+ * @param[in] data     value to be written
+ */
+void at24cxxx_write_byte(const at24cxxx_t *dev, uint32_t pos,  uint8_t data);
+
+/**
+ * @brief Sequentially write \p len bytes from a given position \p pos
+ *
+ * errno is set to ERANGE if EEPROM bounds would be exceeded.
+ * Writing is performed chunk wise, as defined with
+ * AT24CXXX_WRITE_CHUNK_SIZE. The biggest chunk size is the
+ * page size.
+ *
+ * @param[in] dev       AT24CXXX device handle
+ * @param[in] pos       position in EEPROM memory
+ * @param[in] data      write buffer
+ * @param[in] len       requested length to be written
+ *
+ * @return    number of bytes that have been written
+ * @return    0 on failure
+ */
+size_t at24cxxx_write(const at24cxxx_t *dev, uint32_t pos, const uint8_t *data,
```

Comment:
`const void *data` makes for an easier to use API.

Reply:
Ok